env-var-docs/generate_markdown.py

```diff
@@ -0,0 +1,155 @@
+"""Generates markdown from an environment variable documentation file and write it to a github repository.
+
+Requires the following variables to be present in the environment:
+    ENV_VAR_DOCS_PATH: the path to env-var-docs.json.
+
+If LOCAL_OUTPUT is set, markdown will be written to stdout instead of GitHub.
+
+If LOCAL_OUTPUT is not set, the following variables must be set:
+    RELEASE_VERSION: CiviForm version being released. 
+    GITHUB_ACCESS_TOKEN: personal access token to call github API with.
+    TARGET_REPO: the target github repository in 'owner/repo-name' format.
+    TARGET_PATH: the relative path within TARGET_REPO to write the file to.
+"""
+
+import dataclasses
+import env_var_docs.parser
+import env_var_docs.errors_formatter
+import github
+import os
+import sys
+import typing
+
+
+def errorexit(msg):
+    """Logs a message and exits"""
+    sys.stderr.write(msg + "\n")
+    exit(1)
+
+
+@dataclasses.dataclass
+class Config:
+    docs_path: str
+    version: str
+    local_output: bool
+    access_token: str
+    repo: str
+    repo_path: str
+
+
+def make_config() -> Config:
+    """Returns a Config by reading variables set in the environment. See docs
+    at the beginning of this file for required and optional variables.
+
+    Exits if any there are any validation errors.
+    """
+    if "ENV_VAR_DOCS_PATH" not in os.environ:
+        errorexit(
+            "ENV_VAR_DOCS_PATH must be present in the environment variables")
+
+    docs = os.environ["ENV_VAR_DOCS_PATH"]
+    if not os.path.isfile(docs):
+        errorexit(f"'{docs}' does not point to a file")
+
+    local = (os.environ.get("LOCAL_OUTPUT", "false") == "true")
+    if not local:
```

Comment:
**nitpick:** remove `not` and switch the cases since there's an `else` branch

Reply:
Done